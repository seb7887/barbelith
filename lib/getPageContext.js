import { SheetsRegistry } from 'jss';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
import createGenerateClassName from '@material-ui/core/styles/createGenerateClassName';

// App color theme
const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
    suppressDeprecationWarnings: true,
    fontFamily: ['"Segoe UI"', 'sans-serif']
  },
  palette: {
    primary: {
      main: '#fff'
    },
    secondary: {
      main: '#000'
    },
    favoriteIcon: {
      main: '#fc766a'
    },
    type: 'light'
  }
});

function createPageContext() {
  return {
    theme,
    // this is needed in order to deduplicate the injection of css in the page
    sheetsManager: new Map(),
    // this is needed in order to inject the critical css
    sheetsRegistry: new SheetsRegistry(),
    // the standard class name generator
    generateClassName: createGenerateClassName()
  };
}

export default function getPageContext() {
  // make sure to create a new context for every server-side request so that data
  // isn't shared between connections (really really BAD)
  if (!process.browser) {
    return createPageContext();
  }

  // reuse context on the client-side
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
