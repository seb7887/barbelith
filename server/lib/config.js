require('dotenv').config();

const dev = process.env.NODE_ENV !== 'production';
const port = process.env.PORT || 3000;

const config = {
  port: process.env.PORT || 3000,
  db: dev ? process.env.DEV_DB : process.env.MONGO_URI,
  url: dev ? `http://localhost:${port}` : process.env.PROD_URL,
  secret: process.env.SESSION_SECRET || 'jwtSecret'
};

module.exports = config;
