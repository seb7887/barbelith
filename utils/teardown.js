use barbelith-db;
show collections;
db.users.remove({});
db.sessions.remove({});
db.posts.remove({});
