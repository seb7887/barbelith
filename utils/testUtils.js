import faker from 'faker';

export const validUser = () => ({
  name: 'kingmob',
  email: 'king@mob.com',
  password: 'km1234'
});

export const fakeUser = () => ({
  name: faker.internet.userName(),
  email: faker.internet.email(),
  password: faker.internet.password()
});

export const mockedUser = () => ({
  _id: '7',
  avatar: '/static/images/avatar.jpg',
  createdAt: '2019-02-1234',
  email: faker.internet.email(),
  followers: ['1', '2', '3'],
  following: ['3'],
  name: 'Test',
  updatedAt: '2019-02-1234'
});

export const updatedUser = () => ({
  _id: '7',
  name: 'kingmob',
  email: faker.internet.email(),
  avatar: faker.image.avatar(),
  about: faker.lorem.text()
});

export const generateUsers = quantity => {
  const generatedUsers = [];

  for (let i = 0; i < quantity; i++) {
    const user = {
      _id: `${i}`,
      name: `test${i}`,
      avatar: faker.image.avatar()
    };
    generatedUsers.push(user);
  }

  return generatedUsers;
};

export const generateComments = quantity => {
  const generatedComments = [];

  for (let i = 0; i < quantity; i++) {
    const comment = {
      _id: faker.random.uuid().toString(),
      postedBy: {
        _id: '7',
        name: 'test',
        avatar: faker.image.avatar()
      },
      text: faker.lorem.text()
    };
    generatedComments.push(comment);
  }
  return generatedComments;
};

export const fakePost = () => ({
  _id: '7',
  postedBy: {
    _id: '7',
    name: 'test',
    avatar: faker.image.avatar()
  },
  createdAt: faker.date.recent().toString(),
  text: faker.lorem.text(),
  image: faker.image.avatar(),
  likes: ['7'],
  comments: []
});
