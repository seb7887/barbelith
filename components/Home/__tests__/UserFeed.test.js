/* eslint-env jest */
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';

import { render, fireEvent, waitForElement } from 'react-testing-library';
import { generateUsers } from '../../../utils/testUtils';
import { getUserFeed as mockedGetUserFeed } from '../../../lib/api';
import { followUser as mockedFollowUser } from '../../../lib/api';

import UserFeed from '../UserFeed';

jest.mock('../../../lib/api', () => {
  return {
    getUserFeed: jest.fn(() => Promise.resolve()),
    followUser: jest.fn(() => Promise.resolve())
  };
});

jest.mock('next/link', () => {
  return ({ children }) => {
    return children;
  }
});

afterEach(() => {
  mockedGetUserFeed.mockClear();
  mockedFollowUser.mockClear();
});

const users = generateUsers(3);

const auth = {
  user: {
    _id: '7'
  }
};

describe('<UserFeed/>', () => {
  it('renders and matches snapshot', () => {
    mockedGetUserFeed.mockResolvedValueOnce(users);
    const { asFragment } = render(
      <UserFeed auth={auth} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('lists user feed', async () => {
    mockedGetUserFeed.mockResolvedValueOnce(users);
    const { getByTestId, container } = render(
      <UserFeed auth={auth} />
    );

    // Wait for users list
    await waitForElement(() => getByTestId('list'));

    const items = container.querySelectorAll('li');
    expect(items).toHaveLength(users.length);
  });

  it('follows an user from the user feed', async () => {
    mockedGetUserFeed.mockResolvedValueOnce(users);
    mockedFollowUser.mockResolvedValueOnce(users[0]);
    const { getByTestId, container } = render(
      <UserFeed auth={auth} />
    );

    // Wait for users list
    await waitForElement(() => getByTestId('list'));

    const items = container.querySelectorAll('li');
    expect(items).toHaveLength(users.length);

    const followButton = getByTestId('follow');
    fireEvent.click(followButton);

    const snack = await waitForElement(() => getByTestId('snack'));
    expect(snack.textContent).toContain('Following');
  });
});
