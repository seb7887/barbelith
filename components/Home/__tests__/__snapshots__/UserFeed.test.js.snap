// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserFeed/> renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="UserFeed-root-1"
  >
    <h2
      class="MuiTypography-root-6 MuiTypography-h6-23 MuiTypography-alignCenter-29"
      type="title"
    >
      Who to Follow
    </h2>
    <hr
      class="MuiDivider-root-42"
    />
    <ul
      class="MuiList-root-47 MuiList-padding-48"
      data-testid="list"
    />
  </div>
</DocumentFragment>
`;
